waiter/src/waiter/simulator.clj

```diff
@@ -78,104 +87,129 @@
         remaining-completing-requests (- completing-requests completing-request-clients-to-kill)
         clients-to-kill (- clients-to-kill completing-request-clients-to-kill)
         idle-clients-to-kill (min (count (filter #(< tick %) idle-clients)) clients-to-kill)
         clients-to-kill (- clients-to-kill idle-clients-to-kill)
         idle-clients' (vec (concat (drop idle-clients-to-kill (filter #(< tick %) idle-clients))
-                                   (map (fn [_] (+ tick (randomized idle-ticks)))
+                                   (map (fn [_] (+ tick (generate-data-fn idle-ticks)))
                                         (range (+ new-clients remaining-completing-requests)))))
         instances-to-kill (if (neg? scale-amount) (* scale-amount -1) 0)
         starting-servers-to-kill (min (count (filter #(< tick %) starting-servers)) instances-to-kill)
         idle-servers-to-kill (max 0 (- instances-to-kill starting-servers-to-kill))
         remaining-starting-servers (drop-last starting-servers-to-kill starting-servers)
         starting-servers' (vec (concat (filter #(< tick %) remaining-starting-servers)
                                        (if (pos? scale-amount)
-                                         (map (fn [_] (+ tick (randomized startup-ticks)))
+                                         (map (fn [_] (+ tick (generate-data-fn startup-ticks)))
                                               (range scale-amount)) [])))
         servers-started (count (filter #(= tick %) remaining-starting-servers))
         eligible-servers (+ idle-servers completing-requests servers-started)
         available-servers (max 0 (- eligible-servers idle-servers-to-kill))
         activating-clients-to-kill (min activating-clients clients-to-kill)
         remaining-activating-clients (- activating-clients activating-clients-to-kill)
         clients-to-kill (- clients-to-kill activating-clients-to-kill)
-        queued-clients-to-kill (min queued-clients clients-to-kill)
-        remaining-queued-clients (- queued-clients queued-clients-to-kill)
+        queued-clients-to-kill (if clients-exit-immediately
+                                 (min (count queued-clients) clients-to-kill)
+                                 0)
+        remaining-queued-clients (- (count queued-clients) queued-clients-to-kill)
         clients-to-kill (- clients-to-kill queued-clients-to-kill)
         interested-clients (+ remaining-queued-clients remaining-activating-clients)
         requests-created (min interested-clients available-servers)
-        requests-to-cancel (min (count (filter #(< tick %) active-requests)) clients-to-kill)
+        queued-clients' (concat queued-clients (repeat remaining-activating-clients tick))
+        wait-time-max' (->> queued-clients' (map #(- tick %)) (reduce max 0))
+        wait-time-max-ema' (+ (* 0.1 wait-time-max') (* 0.9 wait-time-max-ema))
```

Comment:
What is the purpose of the `wait-time-max-ema`?  It's not in the UI.  Do we still need it?